sdk/core/src/services/WalletStorageService.ts

```diff
@@ -371,15 +365,137 @@ export default class WalletStorageService {
   }
 
-  async deleteCredential(id?: string, credentialIndex?: number): Promise<void> {
-    if ((credentialIndex !== undefined && id) || (!id && credentialIndex === undefined)) {
-      throw new SdkErrorFromCode('COR-1', {
-        errors: [{ message: 'should pass either id or credentialIndex and not both at the same time' }],
+  public async saveCredentials(audienceDid: string, credentials: SignedCredential[], storageRegion?: string) {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+
+    const responses = []
+
+    const encryptedCredentials = await this._encryptCredentials(credentials)
+
+    for (const credential of encryptedCredentials) {
+      const { body } = await this._vaultApiService.storeCredential(token, storageRegion, credential.credentialId, {
+        credentialTypes: credential.credentialTypes,
+        payload: credential.payload,
+      })
+
+      responses.push(body)
+    }
+
+    return responses
+  }
+
+  public async getAllCredentials(audienceDid: string, types: string[][], storageRegion?: string) {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    // convert each type to hash
+    const encryptedTypes: string[][] = []
+    for (const subset of types) {
+      const encryptedSubset: string[] = []
+      for (const type of subset) {
+        const encryptedType = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, type)
+        encryptedSubset.push(encryptedType)
+      }
+      encryptedTypes.push(encryptedSubset)
+    }
+
+    const { body } = await this._vaultApiService.searchCredentials(token, storageRegion, encryptedTypes)
+    const credentials = await this._decryptCredentials(body.credentials)
+
+    // should be deleted during migration to affinidi-vault Phase #2
+    const bloomCredentials = await this.fetchAllDecryptedCredentials()
+    if (types.length === 0) {
+      credentials.push(...bloomCredentials)
+      return credentials
+    }
+
+    for (const credential of bloomCredentials) {
+      const credentialTypes = Array.isArray(credential?.type) ? (credential.type as string[]) : []
+
+      const isFits = types.some((subtypes) => subtypes.every((subtype) => credentialTypes.includes(subtype)))
+
+      if (isFits) {
+        credentials.push(credential)
+      }
+    }
+
+    return credentials
+  }
+
+  public async getCredentialById(audienceDid: string, credentialId: string, storageRegion?: string): Promise<any> {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const hashedId = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+    let encryptedCredential: string
```

Comment:
Extracting this into a separate (private) method `getEncryptedCredentialById` might make code simpler.

(Try to get it from Affinidi Vault, return it; if there was an error: throw it unless it was `AVT-2`, in which case try to get it from Bloom Vault and return it.)